tools/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+add_library( freertos_plus_tcp_tools STATIC )
+
+target_sources( freertos_plus_tcp_tools
+  PRIVATE
+    tcp_utilities/include/tcp_dump_packets.h
+    tcp_utilities/include/tcp_mem_stats.h
+    tcp_utilities/include/tcp_netstat.h
+
+    tcp_utilities/tcp_dump_packets.c
+    tcp_utilities/tcp_mem_stats.c
+    tcp_utilities/tcp_netstat.c
+)
+
+# Note: Have to make system due to compiler warnings in header files.
```

Comment:
Which warnings are these? Can the includes by fixed instead?

Reply:
Yes - the plan here is to add the necessary CMake compilation with minimal changes to the C files themselves and Identify all of the compilation warnings first.
One of them that was identified is `-Wpadded (clang)` - this means that there is additional padding on particular structs etc.  which in most cases is okay, but points to possible improvements for better layout in memory and/or alignment.

A follow on PR for actually removing these warnings will happen after this PR is merged.  I didn't want to have too much in each of the PR's.  What is currently there has been used/consumed for the longest time so assuming none of these are major issues.

Reply:
If you want to see the warnings/changes - they are here:

* https://github.com/phelter/FreeRTOS-Plus-TCP/compare/feature/cmake-support...feature/fixing-clang-gnu-compiler-warnings

The big sets are:
* misuse of doxygen comments or out of date parameter names 
* uninitialized variables at the beginning - which have cases where they may be used before loaded/set - Undefined Behaviour.
* invalid use of unsigned vs. signed - missing casts.

Or you can compile as mentioned with clang-13 or 14 or above and check the warning lists yourselves.

Reply:
Thanks @phelter.  CC: @kstribrnAmzn, @shubnil